auth/oidc.go

```diff
@@ -452,6 +458,35 @@ func formatUsername(value interface{}) (string, error) {
 	}
 }
 
+// getJWTClaimAsSet looks up the given claim in the identity's claims, and transforms it to a set of strings.
+// The claim can be either a string or a slice of strings; other types will result in an error.
+// If the claim is missing from the identity, the empty set (and nil error) will be returned.
+func getJWTClaimAsSet(identity *Identity, claim string) (base.Set, error) {
+	val, ok := identity.Claims[claim]
+	if !ok {
+		return base.Set{}, nil
+	}
+
```

Comment:
Can we use the existing base.ValueToStringArray here, and then call base.SetFromArray on the response?  It seems like it would avoid some duplication on the underlying functionality, and it doesn't look like the approach below would add significant efficiency.

Reply:
Will do in the next commit. The only functional difference I can see is that we'd get more detailed error messages (currently `getJWTClaimAsSet` will tell you exactly which index was a non-string, whereas all we get from `ValueToStringArray` is the values of the non-strings).